src/lib/builders/tree/create.ts

```diff
@@ -0,0 +1,422 @@
+import { get, writable, type Writable } from 'svelte/store';
+import type { Defaults } from "$lib/internal/types";
+import {
+    addEventListener,
+    builder,
+    createElHelpers,
+    executeCallbacks,
+    generateId,
+    getElementByMeltId,
+    isBrowser,
+    isHTMLElement,
+    kbd,
+    styleToString
+} from '$lib/internal/helpers';
+
+import type { CreateTreeViewArgs, ItemDescription, TreeParts } from './types';
+import { onMount, tick } from 'svelte';
+
+
+const defaults = {
+    forceVisible: false,
+} satisfies Defaults<CreateTreeViewArgs>;
+
+const ATTRS = {
+    TABINDEX: 'tabindex',
+    EXPANDED: 'aria-expanded',
+    LABELLEDBY: 'aria-labelledby',
+    DATAID: 'data-id'
+}
+
+const { name } = createElHelpers<TreeParts>('tree-view');
+
+export function createTreeViewBuilder(args: CreateTreeViewArgs) {
+    const argsWithDefaults = { ...defaults, ...args };
+    const { forceVisible } = argsWithDefaults;
+
+    /**
+     * Track currently focused item in the tree.
+     */
+    const focusedItem: Writable<HTMLLIElement | null> = writable(null);
+    const selectedItem: Writable<HTMLLIElement | null> = writable(null);
+    const selectedId: Writable<string | null> = writable(null);
+    const collapsedItems: Writable<string[]> = writable([]);
+
+    let rootEl: HTMLElement | null;
+    let items: HTMLLIElement[];
+    let focusedItemIdx = 0;
+    let itemChildren: ItemDescription[] = [];
+
+    const rootIds = {
+		tree: generateId(),
+		label: generateId(),
+	};
+
+    const rootTree = builder(name(), {
+        returned: () => {
+            return {
+                role: 'tree',
+                'aria-labelledby': rootIds.label,
+                'data-melt-id': rootIds.tree,
+            } as const;
+        }
+    });
+
+    const label = builder(name('label'), {
+        returned: () => ({
+            id: rootIds.label
+        })
+    });
+
+    function updateFocusedElement(newIndex: number) {
```

Comment:
Nice! We have a rovingFocus helper for this, maybe it helps?

Reply:
Awesome, yeah that does the same thing. Using it!😄